cmd/util/ledger/migrations/type_requirements_extractor.go

```diff
@@ -0,0 +1,198 @@
+package migrations
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/old_parser"
+
+	"github.com/onflow/flow-go/cmd/util/ledger/reporters"
+	"github.com/onflow/flow-go/cmd/util/ledger/util/registers"
+	"github.com/onflow/flow-go/fvm/environment"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const typeRequirementExtractingReporterName = "type-requirements-extracting"
+
+type LegacyTypeRequirements struct {
+	typeRequirements []TypeRequirement
+}
+
+type TypeRequirement struct {
+	Address      common.Address
+	ContractName string
+	TypeName     string
+}
+
+func NewTypeRequirementsExtractingMigration(
```

Comment:
Do we need a new migration for this? Could the extraction be integrated into the existing checking migration?

Reply:
I initially had the same thought, but the existing checking migration runs for the new code (i.e: after the staged-contracts migration), but we need to extract the type-requriements from the old code (i.e: before the staged contracts migration)

Reply:
cc: @janezpodhostnik

Reply:
Ah, I see now that this new migration only gathers the contract codes from all registers and extracts the type requirements syntactically. In that case we need a new migration, right. 

Could we maybe still refactor out the common part, the extraction of contract codes from all registers?

Reply:
good idea! added in https://github.com/onflow/flow-go/pull/6341/commits/6736db422e5e4aa2d43e4fb019e06c74217ade64